gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/JsonElementConversionFactory.java

```diff
@@ -0,0 +1,505 @@
+import parquet.schema.Types;
+
+import static org.apache.gobblin.converter.parquet.JsonElementConversionFactory.RecordConverter.RecordType.CHILD;
+import static org.apache.gobblin.converter.parquet.JsonSchema.DataType.*;
+import static parquet.schema.OriginalType.UTF8;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.BINARY;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.INT32;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.INT64;
+import static parquet.schema.Type.Repetition.REPEATED;
+
+
+/**
+ * <p>
+ * Creates a JsonElement to Parquet converter for all supported data types.
+ * </p>
+ *
+ * @author tilakpatidar
+ *
+ */
+public class JsonElementConversionFactory {
+
+  /**
+   * Use to create a converter for a single field from a parquetSchema.
+   *
+   * @param schemaNode
+   * @param state
+   * @param repeated - Is the {@link Type} repeated in the parent {@link Group}
+   * @return
+   */
+  public static JsonElementConverter getConverter(JsonSchema schemaNode, WorkUnitState state, boolean repeated) {
+
+    DataType fieldType = schemaNode.getType();
+    switch (fieldType) {
+      case INT:
+        return new IntConverter(schemaNode, repeated);
+
+      case LONG:
+        return new LongConverter(schemaNode, repeated);
+
+      case FLOAT:
+        return new FloatConverter(schemaNode, repeated);
+
+      case DOUBLE:
+        return new DoubleConverter(schemaNode, repeated);
+
+      case BOOLEAN:
+        return new BooleanConverter(schemaNode, repeated);
+
+      case STRING:
+        return new StringConverter(schemaNode, repeated);
+
+      case ARRAY:
+        return new ArrayConverter(schemaNode, state);
+
+      case ENUM:
+        return new EnumConverter(schemaNode, state);
+
+      case RECORD:
+        return new RecordConverter(schemaNode, state);
+
+      case MAP:
+        return new MapConverter(schemaNode, state);
+
+      default:
+        throw new UnsupportedOperationException(fieldType + " is unsupported");
+    }
+  }
+
+  /**
+   * Converts a JsonElement into a supported ParquetType
+   * @author tilakpatidar
+   *
+   */
+  public static abstract class JsonElementConverter {
+    protected final boolean repeated;
+    protected final JsonSchema jsonSchema;
+    private PrimitiveTypeName targetType;
+
+    /**
+     * HashMap to convert {@link DataType} to {@link PrimitiveTypeName}
+     */
+    private final static HashMap<DataType, PrimitiveTypeName> typeMap = new HashMap<>();
+
+    static {
+      typeMap.put(INT, INT32);
+      typeMap.put(LONG, INT64);
+      typeMap.put(FLOAT, PrimitiveTypeName.FLOAT);
+      typeMap.put(DOUBLE, PrimitiveTypeName.DOUBLE);
+      typeMap.put(BOOLEAN, PrimitiveTypeName.BOOLEAN);
+      typeMap.put(STRING, BINARY);
+    }
+
+    /**
+     * @param jsonSchema
+     * @param repeated
+     * @param targetType
+     */
+    public JsonElementConverter(JsonSchema jsonSchema, boolean repeated, PrimitiveTypeName targetType) {
```

Comment:
I think `targetType` -> `outputType` is a better name? How do you think of that? It follows the conventions used in gobblin `Converter`s like `inputSchema`, `outputSchema`, `inputRecord`, `outputRecord`, etc.

Reply:
Makes sense.